vaadin-crud-flow-parent/vaadin-crud-flow/src/main/java/com/vaadin/flow/component/crud/Crud.java

```diff
@@ -168,8 +175,39 @@ public Crud(Class<E> beanType, CrudEditor<E> editor) {
     public Crud() {
         setI18n(CrudI18n.createDefault(), false);
         registerHandlers();
-        addAttachListener(e -> getElement()
-                .executeJs("this.__validate = function () {return true;}"));
+
+        save = new Button();
+        save.getElement().setAttribute("slot", "save-button");
+        save.addThemeName("primary");
+        getElement().appendChild(save.getElement());
+
+        cancel = new Button();
+        cancel.getElement().setAttribute("slot", "cancel-button");
+        cancel.addThemeName("tertiary");
+        getElement().appendChild(cancel.getElement());
+
+        delete = new Button();
+        delete.getElement().setAttribute("slot", "delete-button");
+        delete.addThemeNames("tertiary", "error");
+        getElement().appendChild(delete.getElement());
+
+        addAttachListener(e -> {
+            getElement()
+                    .executeJs("this.__validate = function () {return true;}");
+            // Override onFormChanges to dispatch an event so the server can
+            // control
+            // the dirty state of the form
+            getElement().executeJs(
+                    "this.__onFormChanges = function () { this.dispatchEvent(new CustomEvent('form-change')); }");
+        });
+
+        getElement().addEventListener("form-change", this::formChangeEvent);
+    }
+
+    private void formChangeEvent(com.vaadin.flow.dom.DomEvent e) {
+        if (this.save.isEnabled()) {
+            this.setDirty(true);
```

Comment:
I think there is a larger issue here: The dirty flag controls several other things:
- whether to ask for confirmation when clicking cancel
- whether to ask for confirmation when editing a different item
- whether to close the editor on outside click or escape

All of these would stop working when disabling the save button, which IMO is unexpected.

Reply:
What is the conclusion on this issue?